### ISSUE TITLE
Different background color for tentative and confirmed bookings

### ISSUE THREAD

**Issue by dyerry79** [meeting-room-booking-system/mrbs-code#3769]
Hello, I need to have a different color for the tentative and confirmed booking. It does seem to work when viewing monthly bookings as seen in the screenshot.

Month View 
![image](https://github.com/user-attachments/assets/1f9136d7-db82-44c5-922c-14a2b48f078c)

However, when the user is under weekly and daily, it doesn't seem to work.

Week View
![image](https://github.com/user-attachments/assets/15b9c16a-1e5b-48f7-8b3a-af11cd7d8a37)

Day View
![image](https://github.com/user-attachments/assets/c4ef951d-c615-4878-8b00-d94c5ecc2338)

I am missing something here for sure. Kindly point me out to the right direction.

**Comment by campbell-m** [comment]
What changes have you made so far?

**Comment by dyerry79** [comment]
> What changes have you made so far?

1. I have this so far for the css. 

.dwm_main .booked a, .dwm_main .booking, .dwm_main .booked span.saving{
    position: relative;
    top: 0;
    left: 0;
    z-index: 50;
    overflow: hidden;
}

.awaiting_approval a::before {
  content: '[For Approval]';
  color:red
}

.awaiting_approval.both {
  background-color: #E1AD01;
  opacity:.9;
}

.awaiting_approval .I{
  background-color: #E1AD01;
}

basically, I thought this solved my issue since I already had these using month, week, and day views. 

month view
![image](https://github.com/user-attachments/assets/e91fb7f9-b09b-4a78-86d2-37ae49e86af8)

week view
![image](https://github.com/user-attachments/assets/12e9f39c-bec1-4222-b520-603aea1b60a1)

day view
![image](https://github.com/user-attachments/assets/c10a9392-3494-4be3-98a6-55c047485d21)

However, once the tentative booking is approved, these are the screens:

month view
![image](https://github.com/user-attachments/assets/079b3a18-bf7e-4787-b53a-c3e2a1f26140)

week view
![image](https://github.com/user-attachments/assets/6b26f65e-7293-4c68-bb87-dcab0d927d21)

day view
![image](https://github.com/user-attachments/assets/5134d263-cdec-43b2-8ce2-5ab06e6c9e7f)

The booking made by the admin has the correct background color while the approved bookings made by non-admin users have the old background for tentative bookings.


2. I have set all the are for this settings
![image](https://github.com/user-attachments/assets/af390dbf-9218-40c8-8af1-7be539ea6c33)

A non-admin user viewing a booking a non-admin user, it works fine, here is the screenshot
![image](https://github.com/user-attachments/assets/6c9554ec-f514-4c13-915a-5d50fe7980c2)

However, a non-admin user viewing a booking done by  an admin user, the private settings seems not working. Here is the screenshot.

![image](https://github.com/user-attachments/assets/a5c611be-dc1b-43c4-89fc-c3e6979667f3)

**Comment by campbell-m** [comment]
I take it you are not using booking types and so don't need to differentiate types by colour?

> However, a non-admin user viewing a booking done by an admin user, the private settings seems not working. Here is the screenshot.

Could you raise this as a separate issue please?

**Comment by dyerry79** [comment]
> I take it you are not using booking types and so don't need to differentiate types by colour?
 
Yes. As I have understood from the testing I made(correct me if I'm wrong), the colors for the booking types are separate from the colors of the approval_status (Tentative and Confirmed). Say if I am going to use the default settings, a **confirmed external[E]** booking type will have a color of cyan while a tentative external[E] booking will still have a cyan color with an opacity of .6.  Here's my test screenshot

![image](https://github.com/user-attachments/assets/a1f93fb3-8ed3-4d79-bc16-6fe7532af208)

Seems also that the $default_type is not working? This is my config settings trying out the booking type for tentative[For Approval] and confirmed bookings.

![image](https://github.com/user-attachments/assets/d404ae62-cbf6-4f67-86c2-ff588dd7c83b)

> Could you raise this as a separate issue please?

Sorry. I'll post this as a separate issue then. 

Thanks.

**Comment by campbell-m** [comment]
First of all, I hope you are overriding the standard CSS by creating a custom CSS file (see systemdefaults.inc.php for how to do this) and not by modifying mrbs.css.php.  That way you won't need to re-apply your changes after MRBS upgrades.  In your custom CSS file you only need to put the changes.  For example, for your first rule you only need

~~~css
.dwm_main .booked a, .dwm_main .booking, .dwm_main .booked span.saving {
  position: relative;
}
~~~

Though I'm curious as to why you need `position: relative;` instead of `position: absolute;`?

I'll look at the main issue later.

**Comment by dyerry79** [comment]
yup, I have a custom css file. 

for the position, this is what I noticed for absolute
![image](https://github.com/user-attachments/assets/3c88a306-0c40-4e99-8860-bd72451806c0)

on the other hand, this is the output of the relative
![image](https://github.com/user-attachments/assets/6ab20e37-7cbb-43f5-914b-848151584f00)

**Comment by dyerry79** [comment]
Yes. Sorry, Confirmed should be Approved. I understand that the booking types is another thing from the approval status. That's the reason I opted to concentrate only on the approval status. Using the custom css, I got what I wanted: Orange for awaiting approval and Green (without opacity) for approved bookings when the user is logged-in to the system. 

On the other hand, if the user is not logged in to the system, the background color is light green. What I am expecting is that approved booking should not have any opacity.

**Comment by campbell-m** [comment]
> On the other hand, if the user is not logged in to the system, the background color is light green. What I am expecting is that approved booking should not have any opacity.

That could be because they are private or tentative bookings.

**Comment by campbell-m** [comment]
Can this be closed now?

**Comment by dyerry79** [comment]
Yes Campbell. Thanks for the guidance.
